include/testharness/TestHarness.h

```diff
@@ -17,7 +18,8 @@ namespace fs = std::filesystem;
 namespace tester {
 
 // Test hierarchy types
-typedef std::vector<std::unique_ptr<TestFile>> SubPackage;
+typedef std::pair<std::unique_ptr<TestFile>, std::optional<TestResult>> TestPair;
```

Comment:
I realize these feel like much the same thing, but this shouldn't be `std::option`, it should be using the promise/future mechanism([std::promise](https://en.cppreference.com/w/cpp/thread/promise), [std::future](https://en.cppreference.com/w/cpp/thread/future)). It has the same sort of mechanism as `std::option` for being in a limbo state with no value, but the outward appearance is immediately apparent to those familiar with the terms. "Option" should be for a value that may or may not have a value, "promise" is for a promised value that _will_ arrive, just in the future, "future" is for getting that promised value. `std::future` also comes with the busy wait mechanism built in at `std::future::wait`.

In particular, `std::optional` should be replaced here with `std::promise`.